File: backend/app/schemas/trainer.py
# backend/app/schemas/trainer.py
from pydantic import BaseModel
from typing import Optional, List # Pastikan List ada di sini
from datetime import date, time # Tambahkan time untuk jadwal

# --- Model yang sudah ada ---
class TrainerBase(BaseModel):
    name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    birth_date: Optional[date] = None
    join_date: Optional[date] = None
    specialization: str
    certification: Optional[str] = None
    hourly_rate: float
    rating: float
    status: Optional[str] = None
    profile_image: Optional[str] = None
    bio: Optional[str] = None

class TrainerCreate(TrainerBase):
    pass

class Trainer(TrainerBase):
    trainer_id: int

    class Config:
        from_attributes = True

class TrainerPerformance(BaseModel):
    id: int
    name: str
    specialization: str
    classes: int
    feedback: float
    retention: float
    activeMembers: int
    status: str
    experience: str

    class Config:
        from_attributes = True

# --- Model Baru untuk Dashboard Data ---
class TrainerStats(BaseModel):
    weekly_classes: int
    active_trainers: int
    high_engagement_classes: int
    avg_satisfaction: float

class TrainerClassParticipantsData(BaseModel):
    trainer: str
    strength: int
    yoga: int
    cardio: int
    pilates: int

class TrainerSatisfactionTrendData(BaseModel):
    week: str
    class Config:
        extra = "allow"

class TrainerClassTypeData(BaseModel):
    name: str
    value: float
    color: str

class TrainerCourseComparisonData(BaseModel):
    type: str
    offline: int
    online: int

class TrainerInsightItem(BaseModel):
    icon_name: str
    title: str
    message: str
    type: str
    color: str

class TrainerAlertItem(BaseModel):
    icon_name: str
    title: str
    message: str
    action: str
    priority: str

# Model Utama untuk semua data Dashboard Trainer
class TrainerDashboardData(BaseModel):
    stats: TrainerStats
    classParticipantsData: List[TrainerClassParticipantsData]
    satisfactionTrendData: List[TrainerSatisfactionTrendData]
    classTypeData: List[TrainerClassTypeData]
    courseComparisonData: List[TrainerCourseComparisonData]
    trainerPerformanceData: List[TrainerPerformance]
    insights: List[TrainerInsightItem]
    alerts: List[TrainerAlertItem]

    class Config:
        from_attributes = True

# --- Skema Baru untuk Trainer Detail Page ---

class TrainerActivityDataItem(BaseModel):
    date: str # Akan dalam format YYYY-MM-DD
    kehadiran: int
    kepuasan: float
    engagement: float

    class Config:
        from_attributes = True

class TrainerScheduleClassItem(BaseModel):
    id: int # schedule_id
    name: str # class name
    time: str # "HH:MM - HH:MM"
    duration: str # "XX min"
    location: str
    participants: str # "X/Y"
    available: int
    type: str # Class type, e.g., "Strength"
    day_of_week: str # e.g., "Senin", "Selasa"

    class Config:
        from_attributes = True

File: src/types/trainer.ts
export interface Trainer {
  trainer_id: number
  name: string
  email?: string
  phone?: string
  birth_date?: string // date akan diterima sebagai string
  join_date?: string // date akan diterima sebagai string
  specialization: string
  certification?: string
  hourly_rate: number
  rating: number
  status?: string
  profile_image?: string
  bio?: string
}

export interface TrainerPerformanceData {
  id: number
  name: string
  specialization: string
  classes: number
  feedback: number
  retention: number
  activeMembers: number
  status: string
  experience: string
}

export interface TrainerStats {
  total_classes_overall: number // Mengganti weekly_classes
  total_trainers_overall: number // Mengganti active_trainers
  high_engagement_classes: number
  avg_satisfaction: number
}

export interface TrainerClassParticipantsData {
  trainer: string
  strength: number
  yoga: number
  cardio: number
  pilates: number
}

export interface TrainerSatisfactionTrendData {
  week: string
  [key: string]: string | number // PERBAIKAN: Sekarang mengizinkan string (untuk 'week') atau number
}

export interface TrainerClassTypeData {
  name: string
  value: number
  color: string
}

export interface TrainerCourseComparisonData {
  type: string
  offline: number
  online: number
}

export interface TrainerInsightItem {
  icon_name?: string
  title: string
  message: string
  recommendation?: string // Solusi bisnis
  type: string // e.g., "trend", "issue", "strength", "opportunity"
  impact: string // e.g., "positive", "negative", "neutral"
  confidence: number // Confidence score dari AI
  color?: string
}

export interface TrainerAlertItem {
  icon_name?: string
  title: string
  message: string
  action?: string
  priority: string
}

export interface TrainerDashboardData {
  stats: TrainerStats
  classParticipantsData: TrainerClassParticipantsData[]
  satisfactionTrendData: TrainerSatisfactionTrendData[]
  classTypeData: TrainerClassTypeData[]
  courseComparisonData: TrainerCourseComparisonData[]
  trainerPerformanceData: TrainerPerformanceData[]
  insights: TrainerInsightItem[]
  alerts: TrainerAlertItem[]
}

// --- Interface Baru untuk Trainer Detail Page ---
export interface TrainerActivityDataItem {
  date: string // Format YYYY-MM-DD atau sesuai yang dikirim backend
  kehadiran: number
  kepuasan: number
  engagement: number
}

export interface TrainerScheduleClassItem {
  id: number // schedule_id
  name: string // class name
  time: string // "HH:MM - HH:MM"
  duration: string // "XX min"
  location: string
  participants: string // "X/Y"
  available: number
  type: string // Class type, e.g., "Strength"
  day_of_week: string // e.g., "Senin", "Selasa"
}

// Skema untuk endpoint AI Insights terpisah
export interface AIInsightsAndAlertsResponse {
  insights: TrainerInsightItem[]
  alerts: TrainerAlertItem[]
}